### PR TITLE
Implemented restricted snippet search within package

### LINKED ISSUE #3: Restrict snippet search within package
For instance:
```
jet> pkg:DataFrames cmd:create
```
Currently only one of those work.

### PR DESCRIPTION
**Pull request by vcoutasso**
Resolves #3

### CODE REVIEW

File: src/actions.jl [hunk 1]
```diff
@@ -71,5 +89,22 @@ function jet(str, should_print=false)
...
+    first = findfirst(':', str)
+    last = findlast(':', str)
+    sep = findlast(' ', str[1:last])
+
+    kind = [str[1:first-1], str[sep+1:last-1]]
+    pkg, keyword = [str[first+1:sep-1], str[last+1:end]]
+
+    if kind[1] in ["pkg", "package"] &&
+       kind[2] in ["cmd", "command", "snippet"]
+      jet_snippet_in(keyword, pkg, should_print)
```
Comment (abelsiqueira) [inline src/actions.jl:103]:
```suggestion
    kws = [str[first+1:sep-1], str[last+1:end]]

    if kind[1] in ["pkg", "package"] &&
       kind[2] in ["cmd", "command", "snippet"]
      jet_snippet_in(keyword, kws[1], kws[2])
    elseif kind[2] in ["pkg", "package"] &&
           kind[1] in ["cmd", "command", "snippet"]
      jet_snippet_in(keyword, kws[2], kws[1])
```
This was we can accept `pkg:... cmd:...` and `cmd:... pkg:...` as well.

**Review by abelsiqueira — CHANGES_REQUESTED** [review]
This is great, thanks. I made a small modification, and I would also ask that you write a simple test in "Basic functionality". In may need to add a new string to "output.jl".

**Comment by vcoutasso** [comment]
Now that the order of `cmd` and `pkg` do not matter, I was wondering what should be the error message in case anything goes wrong. Before that, I tried to enforce the correct order of actions, but I don't think it makes any more sense after these modifications. 

What are your thoughts?

**Comment by abelsiqueira** [comment]
Your observation is correct, the error on line 108 throws when the kinds are not right. So it works for `cmd:... cmd:...` as well as `potato:... tomato:...`. So the current error looks good. We can test these cases, of course.

Regarding your test `pkg:Jet cmd:}`, I fear it's not robust, because `cmd:}` already returns only one entry, so maybe something more generic like `pkg:Jet cmd:the`. Thanks again.

**Comment by abelsiqueira** [comment]
Thanks a lot!

**Comment by abelsiqueira** [comment]
@all-contributors please add @vcoutasso for code

**Pull request merged**